src/silly_sprites/opengl/window.hpp

```diff
@@ -8,11 +8,15 @@
 #include <spdlog/spdlog.h>
 #include <tl/expected.hpp>
 
+namespace sly {
+    class Input;
+}
+
 namespace sly::gl {
 
     class Window final {
     private:
-        friend class Input;
+        friend class ::sly::Input;
         GlfwContext m_context;
 
         using Deleter = void (*)(GLFWwindow*);
```

Comment:
I would prefer this:
```cpp
namespace sly {
    class Input;

    namespace gl {
        class Window final {
            // ...
        };
    } // namespace gl
} // namespace sly
```
What do you think?

Reply:
i like extra namespace for forward declarations.